src/components/Slider/Slider.tsx

```diff
@@ -1,38 +1,65 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { Swiper, SwiperSlide } from 'swiper/react';
+import { Dispatch } from 'redux';
+import { useDispatch } from 'react-redux';
 import SwiperCore, { Autoplay, Navigation } from 'swiper';
 import { Container } from '@mui/material';
 import { SliderCard } from './components/SliderCard';
 import { SliderArrow } from './components/SliderArrow';
-import type { Slides } from './types/SliderTypes';
-import { slides } from './mocks/slidesData';
 import { ArrowDirection } from './constants/constants';
+import { Endpoints } from '../../shared/api/constants/endpoints';
+import { SliderAnyActions, Slides } from './types/SliderTypes';
+import { getSlides } from './services/getSlides';
+import { useSelectorTyped } from './redux/store';
 import 'swiper/swiper-bundle.css';
 import './Slider.scss';
 
 SwiperCore.use([Navigation, Autoplay]);
 
-export const Slider = () => (
-  <Container className="slider-container" maxWidth="xl">
-    <SliderArrow direction={ArrowDirection.PREVIOUS} />
-    <Swiper
-      loop
-      className="slider-box"
-      simulateTouch={false}
-      spaceBetween={10}
-      slidesPerView={1}
-      autoplay={{ delay: 3000, disableOnInteraction: false }}
-      navigation={{
-        prevEl: '.slider-card-icon-previous-arrow',
-        nextEl: '.slider-card-icon-next-arrow',
-      }}
-    >
-      {slides.map((card: Slides, index: number) => (
-        <SwiperSlide key={index}>
-          <SliderCard name={card.name} image={card.imageURL} date={card.date} />
-        </SwiperSlide>
-      ))}
-    </Swiper>
-    <SliderArrow direction={ArrowDirection.NEXT} />
-  </Container>
-);
+export const Slider = () => {
+  const { slides, error, isLoading } = useSelectorTyped(state => state.slider);
+  const dispatch = useDispatch<Dispatch<SliderAnyActions>>();
+
+  useEffect(() => {
+    getSlides(
+      Endpoints.GO_FOR_LAUNCHES_ENDPOINT,
+      Endpoints.PREVIOUS_LAUNCHES_ENDPOINT,
+    )(dispatch);
+  }, []);
+
+  return !error ? (
```

Comment:
The nesting of ternary operators is not very readable. It makes sense for one of them to move to if condition before the  main return